src/gen/lib/Simplify_grammar.ml

```diff
@@ -151,7 +151,127 @@ let translate_named_prec_level translate_name (x : named_prec_level) =
 let translate_precedences translate_name ll =
   List.map (List.map (translate_named_prec_level translate_name)) ll
 
+(* Handle extras that appear in ordinary rules.
+ *
+ * This addresses the issue outlined in
+ * https://github.com/returntocorp/ocaml-tree-sitter-core/issues/36.
+ *
+ * If an extra appears in an ordinary rule, it is, in the general case,
+ * impossible for ocaml-tree-sitter to tell from the CST whether an extra should
+ * be discarded or whether it should be incorporated into the typed CST.
+ *
+ * This addresses that ambiguity by aliasing all extras that appear in other
+ * rules so that they appear with a different name. This way, we can distinguish
+ * between ordinary extras and those that are part of the tree: ordinary extras
+ * will have the original name, and important extras will have the new, aliased
+ * name.
+*)
+let alias_extras grammar =
+  let is_extra =
+    let table =
+      grammar.extras
+      |> List.to_seq
+      |> Seq.filter_map (function
+        | SYMBOL name -> Some name
+        | _ -> None)
+      |> Seq.map (fun name -> (name, ()))
+      |> Hashtbl.of_seq
+    in
+    fun x -> Hashtbl.mem table x
+  in
+  (* Makes an unused name based on a desired target name. Distinct from
+   * Protect_ident which transforms existing names while avoiding collisions. *)
+  let rec make_rule_name =
+    let used_rule_names =
+      grammar.rules
+      |> List.to_seq
+      |> Seq.map (fun (name, _) -> (name, ()))
+      |> Hashtbl.of_seq
+    in
+    fun name ->
+      if Hashtbl.mem used_rule_names name then
+        make_rule_name (name ^ "_")
+      else (
+        Hashtbl.add used_rule_names name ();
+        name
+      )
```

Comment:
The code above looks correct but it can be tricky if we start modifying it. The `Fresh` module will cover this. It's hard to discover, though. Maybe we should have a subfolder named `naming` or some other mechanism that helps us discover which naming techniques are already available.

Reply:
Thanks! I spent a little time looking for something like this but didn't find it. I'll update this to use `Fresh`.